core-bundle/src/Controller/AbstractBackendController.php

```diff
@@ -82,7 +82,15 @@ public function __invoke(): array
             $parameters = [...$getBackendContext(), ...$parameters];
         }
 
-        return parent::render($view, $parameters, $response);
+        $response = parent::render($view, $parameters, $response);
+
+        // Set the status code to 422 if a widget did not validate, so that Turbo can
+        // handle form errors.
+        if ($this->container->get('request_stack')->getMainRequest()->attributes->has('_contao_widget_error')) {
```

Comment:
```suggestion
        if ($this->container->get('request_stack')->getMainRequest()?->attributes->has('_contao_widget_error')) {
```

In case the request stack is empty.